translations/translator.py

```diff
@@ -58,44 +111,120 @@ def __init__(self, files):
 
     def parse_to_dict(self, files):
         for file in files:
-            with open(file) as f:
-                dictdump = json.loads(f.read())
-            self.dict_representation[file] = dictdump
+            self.dict_representation[file] = ResourceFileHandler.read_json(file)
 
     def get_as_dictionary(self):
-        self.parse_to_dict(files)
+        self.parse_to_dict(self.files)
         return self.dict_representation
 
+
 class PropertiesProcessor(object):
     '''
         Processor that can convert a properties file to a python
         dictionary keyed with the file name
     '''
     separator = '='
-    comment_char='#'
+    comment_char = '#'
 
     def __init__(self, files):
         self.files = set(files)
         self.dict_representation = {}
 
     def parse_to_dict(self, files):
         for file in files:
-            current_file_key_val = {}
-            with open(file, "rt") as f:
-                for line in f:
-                    l = line.strip()
-                    if l and not l.startswith(self.comment_char):
-                        key_value_split = l.split(self.separator)
-                        key = key_value_split[0].strip()
-                        value = self.separator.join(key_value_split[1:]).strip().strip('"')
-                        if value:
-                            current_file_key_val[key] = value
-            self.dict_representation[file] = current_file_key_val
+            self.dict_representation[file] = ResourceFileHandler.read_properties(file)
 
     def get_as_dictionary(self):
         self.parse_to_dict(self.files)
         return self.dict_representation
 
+
+class ResourceFileHandler:
+
+    @staticmethod
+    def read(resource_path):
+        extension = resource_path[resource_path.rfind('.') + 1:]
+        if extension == 'properties':
```

Comment:
These checks have already happened in the Validator and the Bundler and a few other places. I think this can be safely removed

Reply:
I'll check